src/lib.rs

```diff
@@ -509,23 +509,31 @@ impl<T> NodeData<T> {
 
 impl<T> Drop for NodeData<T> {
     fn drop(&mut self) {
-        // Collect all descendant nodes and detach them to prevent the stack overflow.
```

Comment:
I think you should keep this comment. Otherwise it's not clear why we need a custom Drop.

Reply:
Agreed.